lib/src/entity.dart

```diff
@@ -9,7 +9,7 @@ import 'package:flame_behaviors/flame_behaviors.dart';
 /// The visualization of the entity is defined by the [Component]s that are
 /// attached to it.
 /// {@endtemplate}
-abstract class Entity extends PositionComponent {
+abstract class Entity extends PositionComponent with HasGameRef {
```

Comment:
We should probably add a proxy generics here now, otherwise the entity will not be able to define the specific game class.

```suggestion
abstract class Entity<T extends Game> extends PositionComponent with HasGameRef<T> {
```